lib/iris/fileformats/nimrod_load_rules.py

```diff
@@ -97,186 +348,315 @@ def reference_time(cube, field):
+def coord_system(field):
+    """Define the coordinate system for the field."""
+    ellipsoid = proj_biaxial_ellipsoid(field)
 
-    """
-    # "NG" (British National Grid)
     if field.horizontal_grid_type == 0:
-        british_national_grid_x(cube, field)
-        british_national_grid_y(cube, field)
+        # Check for missing grid meta-data and insert OSGB definitions.
+        # Some Radarnet files are missing these.
+        set_british_national_grid_defaults(field)
+    if field.horizontal_grid_type == 0 or field.horizontal_grid_type == 4:
+        coord_sys = iris.coord_systems.TransverseMercator(
+            field.true_origin_latitude,
+            field.true_origin_longitude,
+            field.true_origin_easting,
+            field.true_origin_northing,
+            field.tm_meridian_scaling,
+            iris.coord_systems.GeogCS(**ellipsoid),
+        )
+    elif field.horizontal_grid_type == 1:
+        coord_sys = iris.coord_systems.GeogCS(**ellipsoid)
     else:
         raise TranslationError(
-            "Grid type %d not yet implemented" % field.horizontal_grid_type
+            "Coordinate system for field type {} not implemented".format(
+                field.horizontal_grid_type
+            )
         )
+    return coord_sys
 
 
-def orography_vertical_coord(cube, field):
-    """Special handling of vertical coords for orography fields: Do nothing."""
-    # We can find values in the vertical coord, such as 9999,
-    # for orography fields. Don't make a vertical coord from these.
-    pass
-
+def horizontal_grid(cube, field):
+    """Add X and Y coordinates to the cube.
 
-def height_vertical_coord(cube, field):
-    """Add a height coord to the cube, if present in the field."""
-    if (
-        field.reference_vertical_coord_type == field.int_mdi
-        or field.reference_vertical_coord == field.float32_mdi
-    ):
-        height_coord = DimCoord(
-            field.vertical_coord,
-            standard_name="height",
-            units="m",
-            attributes={"positive": "up"},
+    """
+    if field.origin_corner != 0:
+        raise TranslationError(
+            "Corner {0} not yet implemented".format(field.origin_corner)
         )
-        cube.add_aux_coord(height_coord)
+    crs = coord_system(field)
+    if field.horizontal_grid_type == 0 or field.horizontal_grid_type == 4:
+        units_name = "m"
+        x_coord_name = "projection_x_coordinate"
+        y_coord_name = "projection_y_coordinate"
+    elif field.horizontal_grid_type == 1:
+        units_name = "degrees"
+        x_coord_name = "longitude"
+        y_coord_name = "latitude"
     else:
-        raise TranslationError("Bounded vertical not yet implemented")
+        raise TranslationError(
+            "Horizontal grid type {} not "
+            "implemented".format(field.horizontal_grid_type)
+        )
+    points = (
+        np.arange(field.num_cols) * field.column_step + field.x_origin
```

Comment:
I don't know how big these things get, but I presume you have no real precision worries here?
This way reads nicely, but np.linspace will often work out better if you might be worried about precision.
(+ same at line 562)

Reply:
Changed to use linspace as I think it is a better solution. Results don't change.

Reply:
Ok.  I see that as you have the necessary values to hand, it looks quite neat anyway.